03/Product.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+const Product = (props) => {
+    const {item, isCart, isCategory, add, remove} = props;
+
+    const button = () => {

```

Comment:
W zasadzie nie trzeba było z tego robić funkcji, do zmiennej można przypisać JSX.
Jak już ma to być funkcja to odpowiednio bym ją nazwał np. `renderButton` - bo to czynność.
Można też zrobić z tego osobny komponent :)